sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -180,20 +177,11 @@ void guessLocalWorkSize(ur_device_handle_t Device, size_t *ThreadsPerBlock,
 // If the kernel requires a number of registers for the entire thread
 // block exceeds the hardware limitations, then the cuLaunchKernel call
 // will fail to launch with CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES error.
-bool hasExceededMaxRegistersPerBlock(ur_device_handle_t Device,
-                                     ur_kernel_handle_t Kernel,
-                                     size_t BlockSize) {
-  int MaxRegsPerBlock{0};
-  UR_CHECK_ERROR(cuDeviceGetAttribute(
-      &MaxRegsPerBlock, CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_BLOCK,
-      Device->get()));
-
-  int RegsPerThread{0};
-  UR_CHECK_ERROR(cuFuncGetAttribute(&RegsPerThread, CU_FUNC_ATTRIBUTE_NUM_REGS,
-                                    Kernel->get()));
-
-  return BlockSize * RegsPerThread > size_t(MaxRegsPerBlock);
-}
+static auto hasExceededMaxRegistersPerBlock = [](ur_device_handle_t Device,
```

Comment:
This has been changed into a lambda member, I know it's only semantic sugar, but a member function seems more appropriate.

Reply:
Alright, that's been changed now